src/tests/ftest/util/test_utils_pool.py

```diff
@@ -19,12 +19,89 @@
 from dmg_utils import DmgCommand
 
 
+def add_pool(test, namespace=None, create=True, connect=True, index=0, **params):
```

Comment:
Yeah, eventually it would be good to remove the `TestWithServers.pool`, `TestWithServers.add_pool()`, and `TestWithServers.get_pool()` as this replaces the need for them , but that is a much bigger PR.